File: src/TimeControl/Models/Reports.cs
using System.Collections.Generic;

namespace TimeControl.Models
{
    public class Reports
    {
        public class Projects
        {
            public List<Project> ListProjects { set; get; }
        }

        public class Workers
        {
            public List<Worker> ListWorkers { set; get; }
        }

        public class ProjectCountsWorkers
        {
            public List<ProjectCountWorkers> ListProjectsCountWorkerses { set; get; }
        }

        public class Project
        {
            public string Name { set; get; }
            public float Time { set; get; }
        }

        public class Worker
        {
            public string Name { set; get; }
            public float Time { set; get; }
        }

        public class ProjectCountWorkers
        {
            public string Name { set; get; }
            public int Count { set; get; }
        }
    }
}

File: src/TimeControl/angular2App/app/models/reports.ts
export class ReportProjects {
    public listProjects: ProjectReportItem[];

    constructor() {
        this.listProjects = [];
    }
}

export class ReportWorkers {
    public listWorkers: WorkerReportItem[];

    constructor() {
        this.listWorkers = [];
    }
}

export class ProjectCountsWorkers {
    public listProjectsCountWorkerses: ProjectCountWorkersItem[];

    constructor() {
        this.listProjectsCountWorkerses = [];
    }
}


export class ProjectReportItem {
    public time: number;
    public name: string;
}

export class WorkerReportItem {
    public time: number;
    public name: string;
}

export class ProjectCountWorkersItem {
    public count: number;
    public name: string;
}

